### PR TITLE
Don't pixel align if the map isn't aligned.

### PR DESCRIPTION
**Pull request by manthey**
If the map is not at an integer offset on the page, don't try to align texels.  This only applies to the vgl renderer.

### CODE REVIEW

File: src/camera.js [hunk 1]
```diff
@@ -303,10 +303,17 @@
...
      * Getter/setter for the viewport.
+     *
+     * The viewport consists of a width and height in pixels, plus a left and
+     * top offset in pixels.  The offsets are only used to determine if pixel
+     * alignment is possible.
      */
     Object.defineProperty(this, 'viewport', {
       get: function () {
-        return {width: this._viewport.width, height: this._viewport.height};
+        return {
```
Comment (aashish24) [inline src/camera.js]: why not just return the viewport object?

File: src/gl/vglRenderer.js [hunk 2]
```diff
@@ -189,8 +189,11 @@ var vglRenderer = function (arg) {
       m_lastZoom = map.zoom();
       cam.setViewMatrix(view, true);
       cam.setProjectionMatrix(proj);
+      var viewport = camera.viewport;
       if (proj[1] || proj[2] || proj[3] || proj[4] || proj[6] || proj[7] ||
           proj[8] || proj[9] || proj[11] || proj[15] !== 1 || !ortho ||
+          (viewport.left && viewport.left % 1) ||
```
Comment (aashish24) [inline src/gl/vglRenderer.js]: some comments would be nice here on reasoning behind 1%

File: src/camera.js (hunk 1 above)
Comment (manthey) [inline src/camera.js]: To prevent a consumer of this from mutating the internal object.

File: src/gl/vglRenderer.js (hunk 2 above)
Comment (manthey) [inline src/gl/vglRenderer.js]: Okay.  I thought it was a familiar way to check if a value was an integer.

File: src/camera.js (hunk 1 above)
Comment (aashish24) [inline src/camera.js]: Sure, could we not use jquery extend in that case (like we use it elsewhere?) but I am fine with this code given your explanation.

File: src/gl/vglRenderer.js (hunk 2 above)
Comment (aashish24) [inline src/gl/vglRenderer.js]: for some but also it would be nice to explain the code block around it if possible.

File: src/camera.js (hunk 1 above)
Comment (manthey) [inline src/camera.js]: In theory, it should be more efficient to do this ourselves than use jQuery's extend, since we know what keys are of interest.

Comment (aashish24) [inline src/camera.js]: ok, sounds good.

**Pull request merged**